### ISSUE TITLE
remove invalid integration test case

### ISSUE THREAD

**Issue by skriss** [projectcontour/contour#3391]
[Part of the 001-required-field-validation](https://github.com/projectcontour/contour/blob/main/_integration/testsuite/httpproxy/001-required-field-validation.yaml#L63-L80) integration test case tries to validate that the `spec.tcpproxy.services` field is required.

However, it's not required: see https://github.com/projectcontour/contour/pull/2684 and https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/httpproxy.go#L461-L463.

This test has been passing because, up through Kubernetes 1.19, attempting to create the `HTTPProxy` linked above generated a related but different error: `The HTTPProxy "missing-tcp-services" is invalid: spec.tcpproxy.services: Invalid value: "null": spec.tcpproxy.services in body must be of type array: "null"`. This error was close enough to make the integration test case pass.

As of Kubernetes 1.20, it no longer generates that error, and allows the `HTTPProxy` to be created, since `spec.tcpproxy.services` is indeed optional. As such, the test case now *fails* on 1.20.

TLDR: we should remove https://github.com/projectcontour/contour/blob/main/_integration/testsuite/httpproxy/001-required-field-validation.yaml#L63-L80 because `spec.tcpproxy.services` is no longer a required field.
